### PR TITLE
Change CORS to middleware pattern

### PR DESCRIPTION
**Pull request by senthalan**
Fix https://github.com/asgardeo/thunder/issues/463

This pull request refactors the CORS middleware implementation in the backend, moving CORS logic out of the generic server operations code and into a dedicated middleware package. It introduces a new, simplified CORS middleware API and comprehensive tests, while removing legacy server operation abstractions and related tests. These changes improve modularity, clarity, and testability of CORS handling.

### CORS Middleware Refactor

* Moved CORS logic from `backend/internal/system/server/serveroperation.go` to a new dedicated middleware package (`backend/internal/system/middleware/cors.go`), renaming and rewriting the file for clarity and separation of concerns.
* Introduced a new `CORSOptions` struct and the `WithCORS` middleware function, replacing the previous `ServerOperationService` abstraction for wrapping handlers with CORS support. Caching of allowed origins is now handled via package-level functions. [[1]](diffhunk://#diff-7f771de31cefb20d2c8bcc1ec44fde3a036faf732e4fd411406938ff5c5c57c7L31-R91) [[2]](diffhunk://#diff-7f771de31cefb20d2c8bcc1ec44fde3a036faf732e4fd411406938ff5c5c57c7L82-L115)
* Added a comprehensive test suite for the new CORS middleware in `backend/internal/system/middleware/cors_test.go`, covering various scenarios and edge cases.

### Removal of Legacy Server Operation Code

* Removed the old `ServerOperationService`, its interface, request wrapping logic, and related CORS configuration types from `backend/internal/system/server/model.go` and deleted the corresponding test file `backend/internal/system/server/serveroperation_test.go`. [[1]](diffhunk://#diff-c39c0a121ee6259fd3c56f88b63012be66470974cba397cb7e5670c9b02a764aL1-L31) [[2]](diffhunk://#diff-d435c3872495b4f03b3254c467e87ccdfa49271673b9d4e629ea47fd2413c6aeL1-L280)

### Integration Updates

* Updated service initialization in `backend/internal/system/services/applicationservice.go` to use the new middleware package instead of the legacy server operations code.

### CODE REVIEW

File: backend/internal/system/middleware/cors.go [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+			w.Header().Set("Access-Control-Allow-Credentials", "true")
+		}
+	}
+}
+
+// getAllowedOrigins retrieves the list of allowed origins from configuration.
+func getAllowedOrigins() []string {
+	logger := log.GetLogger().With(log.String(log.LoggerKeyComponentName, "CORSMiddleware"))
+
+	// TODO: Revisit this when adding support for the organization concept.
```
Comment (ThaminduDilshan) [inline backend/internal/system/middleware/cors.go]:
```suggestion
```
Let's remove this as it is no longer needed
